internal/transport/http2_server.go

```diff
@@ -445,6 +445,9 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 		streamID: s.id,
 		wq:       s.wq,
 	})
+	if s.state == streamReadDone || s.state == streamDone {
```

Comment:
Is it impossible for another thread to be modifying this state's stream here?  I think it might be, but not sure.  You should probably use `getState` just to be paranoid.

Reply:
Got rid of this check, since this was what was prevent the stream from being handled. Stream is now handled on a headers with END_STREAM.